tempto-core/src/main/java/com/teradata/tempto/internal/initialization/RequirementsExpanderInterceptor.java

```diff
@@ -80,6 +88,49 @@ public static int getMethodsCountFromContext(ITestContext context)
         return allExpandedMethods;
     }
 
+    private void incrementSeenMethodsCount(List<IMethodInstance> newExpandedMethods)
+    {
+        for (IMethodInstance newExpandedMethod : newExpandedMethods) {
+            Optional<Object[][]> parametersForMethod = getParametersForMethod(newExpandedMethod);
+            if (parametersForMethod.isPresent()) {
+                seenMethodsCount += parametersForMethod.get().length;
+            } else {
+                seenMethodsCount++;
+            }
+        }
+    }
+
+    private Optional<Object[][]> getParametersForMethod(IMethodInstance method)
+    {
+        Test testAnnotation = method.getMethod().getConstructorOrMethod().getMethod().getAnnotation(Test.class);
+        Class dataProviderClass = testAnnotation.dataProviderClass();
+        if (dataProviderClass == null || dataProviderClass == Object.class) {
+            dataProviderClass = method.getMethod().getRealClass();
+        }
+        String dataProviderName = testAnnotation.dataProvider();
+        if (dataProviderName.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Optional<Method> dataProviderMethod = asList(dataProviderClass.getMethods()).stream().filter(
+                m -> {
+                    DataProvider annotation = m.getAnnotation(DataProvider.class);
+                    return annotation != null && annotation.name().equals(dataProviderName);
+                }
+        ).findFirst();
+        if (dataProviderMethod.isPresent()) {
+            try {
+                return Optional.of((Object[][]) dataProviderMethod.get().invoke(method.getInstance()));
```

Comment:
ah... it is a static method, so what if there are multiple data provider static methods, will you just find first of them?